src/main/java/org/apache/commons/imaging/ImageParser.java

```diff
@@ -123,12 +115,18 @@
                 new TiffImageParser(),
                 new WbmpImageParser(),
                 new XbmImageParser(),
-                new XpmImageParser(),
+                new XpmImageParser()
                 // new JBig2ImageParser(),
                 // new TgaImageParser(),
-        };
+        );
     }
 
+    /**
+     * Get a default parameters instance for this parser.
+     * @return default parameters instance
+     */
+    public abstract T getDefaultParameters();
```

Comment:
Copied from @darkma773r 's PR. By moving it here from the `ImageFormat` (i.e. responsible for defining the default parameter is the parser, not the format) several generics warnings don't happen anymore.